elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexWorkVisitor.java

```diff
@@ -66,6 +68,25 @@
  */
 class ElasticsearchIndexWorkVisitor implements IndexWorkVisitor<Void, BackendRequest<?>> {
 
+	/*
+	 * Nesting marker field name: fields with this name have the parent path as their value,
+	 * and are used for denoting the parent element of the subsequent fields
+	 *
+	 * Use a very specific String to avoid naming conflict with user-defined field names
+	 */
+	public static final String NESTING_MARKER = "__Hibernate_nesting_marker";
+
+	/*
+	 * "this" marker field name: fields with this name are used for denoting that the
+	 * subsequent field represents an embedded as a whole, as opposed to simply a field
+	 * in an embedded.
+	 * The marker is necessary because in some cases, the embedded and its elements
+	 * share a common field name. For instance it is the case of embedded collections.
+	 *
+	 * Use a very specific String to avoid naming conflict with user-defined field names
+	 */
+	public static final String THIS_MARKER = "__Hibernate_this_marker";
```

Comment:
This marker is useful when `@IndexedEmbedded` collections are indexed.

For instance:

```java
@Indexed
public class MyEntity {
@IndexedEmbedded(indexNullAs = "NULL")
private List<String> myCollection;
}
```

When `myCollection` contains `["foo", "bar"]`, the following fields will appear in the Lucene document:

```
myCollection:foo
myCollection:bar
```

When `myCollection` is `null`, the following field will appear:
```
myCollection:NULL
```

When you're in the second case you cannot know only from the Lucene document whether the field represents a collection element (say, `myCollection[1]` in pseudo-code) or the collection itself (still in pseudo-code, `myCollection`, or `myCollection.this` to be more explicit).

"This" or "self" are the only words that came to my mind and seemed to fit the meaning of this marker: it states that the field value is not the value of a collection element, but really the value of the embedded itself.

Be aware, though, that in order to fix HSEARCH-2397 I'm already refactoring this: just using a string for the nesting path is not enough, the marker has to convey more information. So it might not be worth a long discussion...